yellowbrick/features/missing.py

```diff
@@ -0,0 +1,129 @@
+# yellowbrick.features.missing
+# Feature importance visualizer
+#
+# Author:  Nathan Danielsen <nathan.danielsen@gmail.com>
+# Created: Fri Mar 24 8:17:36 2018 -0500
+#
+# Copyright (C) 2018 District Data Labs
+# For license information, see LICENSE.txt
+#
+# ID: missing.py [] nathan.danielsen@gmail.com.com $
+
+"""
+Implementation of missing values visualizers
+
+To Include:
+- Bar
+- Density Matrix (by time, specifiable index)
+- Heatmap
+
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from yellowbrick.utils import is_dataframe
+from yellowbrick.utils import is_structured_array
+from yellowbrick.features.base import DataVisualizer
+
+# from yellowbrick.style.colors import resolve_colors
+
+
+
+##########################################################################
+## Feature Visualizer
+##########################################################################
+
+
+class MissingValuesBarVisualizer(DataVisualizer):
+    """
+    """
+
+    def __init__(self,
+                 ax=None,
+                 x=None,
+                 y=None,
+                 features=None,
+                 classes=None,
+                 color=None,
+                 colormap=None,
+                 **kwargs):
+        """
+        """
+
+        super(MissingValuesBarVisualizer, self).__init__(ax, features, classes, color,
+                                                colormap, **kwargs)
+
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        TODO if y, then color code the missing values in the chart?
+
+
+        """
+        nrows, ncols = df.shape
+
+        # Handle the feature names if they're None.
+        if self.features_ is not None and is_dataframe(X):
+            X = X[self.features_].as_matrix()
+
+        # handle numpy named/ structured array
+        elif self.features_ is not None and is_structured_array(X):
+            X_selected = X[self.features_]
+            X = X_selected.copy().view((np.float64, len(X_selected.dtype.names)))
+
+        else:
+            pass
+
+        if self.features_ is None:
+            self.features_ = range(nrows)
+
+        if self.classes_ is None:
+            # TODO: Is this the most efficient method?
```

Comment:
Yes- I did some benchmarking and np.unique is the best method -- particularly because it returns sorted classes. I'd further recommend turning this into a np.array instead of a list.

Reply:
https://bbengfort.github.io/observations/2017/05/02/python-unique-benchmark.html